frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from '@mui/material';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { getCluster } from '../../lib/cluster';
+import { post } from '../../lib/k8s/apiProxy';
+import Namespace from '../../lib/k8s/namespace';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton, AuthVisible } from '../common';
+
+export function CreateNamespaceButton() {
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [namespaceName, setNamespaceName] = useState('');
+  const [namespaceDialogOpen, setNamespaceDialogOpen] = useState(false);
+  const dispatchCreateEvent = useEventCallback(HeadlampEventType.CREATE_RESOURCE);
+  const dispatch = useDispatch();
+
+  function createNewNamespace() {
+    function validateNamespaceName(namespaceName: string) {
+      const namespaceRegex = /^[a-z]([-a-z0-9]*[a-z0-9])?$/;
+
+      // Check if the namespace name is too long
+      if (namespaceName.length > 63) {
+        alert(t('translation|Invalid namespace name: name must be 63 characters or shorter'));
+        return false;
+      }
+
+      // Check if the namespace name starts with a alphanumeric character
+      const startsAlphanumeric = /^[a-z0-9]/.test(namespaceName);
+      if (!startsAlphanumeric) {
+        alert(
+          t('translation|Invalid namespace name: name must start with an alphanumeric character')
+        );
+        return false;
+      }
+
+      // Check if the namespace name ends with an alphanumeric character
+      const endsAlphanumeric = /[a-z0-9]$/.test(namespaceName);
+      if (!endsAlphanumeric) {
+        alert(
+          t('translation|Invalid namespace name: name must end with an alphanumeric character')
+        );
+        return false;
+      }
+
+      // Check if the namespace name contains only lowercase letters, numbers, or hyphens
+      const validChars = namespaceRegex.test(namespaceName);
+      if (!validChars) {
+        alert(
+          t(
+            'translation|Invalid namespace name: name must consist of lower case alphanumeric characters or -'
+          )
+        );
+        return false;
+      }
+
+      return true;
+    }
```

Comment:
See my comment about the function we have in SettingsCluster.tsx.

Reply:
fixed!